### PR TITLE
Add fusion support for Dnnl execution provider

### PR DESCRIPTION
**Pull request by georgen117**
**Description**:
This will add op fusion detection to the dnnl (OneDNN) execution provider. 

In addition 4 types of fusion are being detected.
- Gelu (gelu_erf)
- FastGelu (gelu_tanh)
- Conv+Relu
- MatMul+Add

The Gelu fusions and MatMul+Add were added because they are patterns that frequently show up in NLP models BERT and GPT2
The Conv+Relu fusion was added because it frequently shows up in Convolution based models like Resnet.

**Motivation and Context**
- Why is this change required? What problem does it solve?
Due to the fact that Dnnl execution provider creates its own subgraphs the OnnxRuntime Optimizer code will not detect fusion patterns for the nodes claimed by the dnnl ep.  This enables detecting fusion patterns that the dnnl ep can support.

### CODE REVIEW

**Comment by georgen117** [comment]
@jywu-msft maintainer please remember to run /azp run Linux DNNL CI Pipeline since this is the main pipeline that runs the dnnl ep.

**Comment by georgen117** [comment]
@jywu-msft now that we have run and passed the dnnl pipeline could you please start the other CI pipelines. Thanks :)

**Comment by georgen117** [comment]
@jywu-msft  I have looked at the failure for the `orttraining-amd-gpu-ci-pipeline` and  I don't see the source or the error. It kinda looks to me there was an error early in the build script but not the exact cause.  @jywu-msft do you have any possible feedback about the source of the failure?

**Comment by jywu-msft** [comment]
> @jywu-msft I have looked at the failure for the `orttraining-amd-gpu-ci-pipeline` and I don't see the source or the error. It kinda looks to me there was an error early in the build script but I can not pinpoint the exact cause. @jywu-msft do you have any possible feedback about the source of the failure?

it doesn't look to be related to your changes. I see the failure in other runs. let me follow up and find out who is working on fixing it. sorry for the inconvenience.

**Comment by jywu-msft** [comment]
> > @jywu-msft I have looked at the failure for the `orttraining-amd-gpu-ci-pipeline` and I don't see the source or the error. It kinda looks to me there was an error early in the build script but I can not pinpoint the exact cause. @jywu-msft do you have any possible feedback about the source of the failure?
> 
> it doesn't look to be related to your changes. I see the failure in other runs. let me follow up and find out who is working on fixing it. sorry for the inconvenience.

that pipeline isn't required at the moment, so it won't block merging of this PR.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**
